### PR TITLE
 #171443039 fix bug on reject specific request by manager

### PR DESCRIPTION
**Pull request by Alexander96779**
#### What does this PR do?
This PR fixes bugs on reject requests by the line manager endpoint.
#### Description of Task to be completed?
Have manager be able to reject a request if it is found in his report
- Check if the user is a manager.
- Check if the request he is trying to reject is found in his direct report
- Check if the request is not rejected already
- Reject request
#### How should this be manually tested?
- You have to clone the repo `git clone https://github.com/Stackup-Rwanda/fan7-bn-backend.git`
- Then After the repo cloned `cd project-folder` and run `npm i` to install all dependencies
- Run app by `npm run dev`
- Use Postman to test the endpoint, then type the following in URL `PATCH: localhost:5000/api/requests/:request_id/reject`
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#171443039](https://www.pivotaltracker.com/story/show/171443039)

### CODE REVIEW

File: src/test/requests.test.js [hunk 1]
```diff
@@ -8,10 +8,10 @@ chai.should();
 describe('Requests Test', () => {
   it('should be able to reject if line manager', done => {
     chai.request(server)
-      .patch('/api/requests/2/reject')
+      .patch('/api/requests/9/reject')
       .set(
         'token',
-        'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZW1haWwiOiJ0cmV5M0BnbWFpbC5jb20iLCJpYXQiOjE1ODEwMDI3NjJ9.FGgqQWqOEYO6ovfvq_bM58pYXY0VG1-3Wm-hQ59_koE'
+        'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NiwiZW1haWwiOiJlbHZpc3J1Z2FtYmFAZ21haWwuY29tIiwicm9sZSI6Im1hbmFnZXIiLCJpYXQiOjE1ODI1NzA3MjF9.HisvoHo0t8cmWvpGokttlXtV07QOqUBB8dq4vqQ-s14'
```
Comment (ajimae) [inline src/test/requests.test.js]: Why are we hardcoding this token?
Can't you generate one by registering a user?

What will happen to your test when this token expires?

**Review by ajimae — COMMENTED** [review]
Please take a look at the comment I left on this PR @Alexander96779 

Thank you.

File: src/test/requests.test.js (hunk 1 above)
Comment (Alexander96779) [reply src/test/requests.test.js]: let me fix it

Comment (Elvis-rugamba) [reply src/test/requests.test.js]: Use before function in your tests to login or signup a user

**Review by Elvis-rugamba — APPROVED** [review]
LGTM

**Review by ajimae — APPROVED** [review]
LGTM!

**Pull request merged**